packages/channels/botpress-channel-messenger/src/index.js

```diff
@@ -56,9 +56,21 @@ const initializeMessenger = async (bp, configurator) => {
     return bp.logger.warn('[Messenger] Connection disabled')
   }
 
+  const page_ids = Object.keys(messenger.config.pages)
   return messenger
-    .connect()
-    .then(() => messenger.updateSettings())
+    .connect(page_ids)
+    .then(() => {
+      if (!config.freezeProfile) {
+        if (page_ids.length > 0) {
+          for (let i = 0; i < page_ids.length; i++) {
+            messenger.updateMessengerProfile(page_ids[i])
+          }
+        } else {
+          messenger.updateMessengerProfile()
+        }
+      }
+      return messenger
```

Comment:
OK, will change to a guard statement in the next commit

Yes, on Facebook a bot always connects to a page. There is no other object/option

Reply:
> Yes, on Facebook a bot always connects to a page. There is no other object/option

If so, I think `pages` should be required and you should always provide at least one page.

Reply:
It would be nice to make `pages` required, but that would require updating configs on all existing installs. In my current PR though it if any setting in `pages` is not set there is a fallback on legacy config 

Legacy config is also useful for settings that are the same across all pages (e.g enabling persistent menu). Only `accessToken` must be unique per page, but it could also fall back to default config if there is only one page

I have tested both cases

Reply:
We could show deprecation warning and only show correct settings examples in readme